### ISSUE TITLE
search.js memory leak

### ISSUE THREAD

**Issue by smurthas** [LockerProject/Locker#226]
I'm not sure what's up, but 4.8GB seems like a bit too much. :)

```
top - 00:43:15 up 10 days, 1 min,  3 users,  load average: 9.49, 10.74, 9.92
Tasks: 509 total,   1 running, 508 sleeping,   0 stopped,   0 zombie
Cpu(s):  6.1%us, 13.3%sy,  0.3%ni, 76.5%id,  0.8%wa,  0.0%hi,  1.0%si,  2.0%st
Mem:   7645964k total,  7525388k used,   120576k free,    11176k buffers
Swap: 15291924k total,  6857216k used,  8434708k free,   574904k cached

  PID USER      PR  NI  VIRT  RES  SHR S %CPU %MEM    TIME+    TIME COMMAND                                                    
27872 ubuntu    20   0 9868m 4.8g 6080 S    0 65.6  35:45.22  35:45 node search.js                                              
21203 ubuntu    20   0  641m 133m 132m S    0  1.8   9:15.30   9:15 mongod --dbpath /var/locker/Me/mongodata --port 27018
```

**Comment by othiym23** [comment]
<pre>
root@ip-xx-xx-xx-xx:/var/lxc-machines/storage/locker_tim_25/Me/search/search.index# ps auxwww | egrep "(PID|27872)"
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
ubuntu   27872 15.5 67.9 17445360 5194708 ?    Sl   Sep24  64:09 node search.js
root@ip-xx-xx-xx-xx:/var/lxc-admin/backup# lsof -p 27872 | wc -l
1002
root@ip-xx-xx-xx-xx:/var/lxc-machines/storage/locker_tim_25/Me/search/search.index# du -sh .
119M    .
</pre>


Also:

<pre>
procs -----------memory---------- ---swap-- -----io---- -system-- ----cpu----
 r  b   swpd   free   buff  cache   si   so    bi    bo   in   cs us sy id wa
 8  0 14012408  67256   5772 365044   99  166  2994   729 1707 7532 45  6 26  6
</pre>

**Comment by amcjen** [comment]
Arrrrrrgh. I bet this has something to do w/ the IndexWriter issue Temas found. 

We'll tackle it next week I think and update the node clucene module, then see wtf is going on. :)

Woo! Good find tho!
E.

**Comment by mdz** [comment]
Whose locker is that?

## 
- mdz

**Comment by smurthas** [comment]
tim c

**Comment by temas** [comment]
This is code complete and pending a staging environment to do one more live test on before we green light it.

**Comment by ctide** [comment]
Tests are broken on that branch - http://ci.lockerproject.org/job/LP2%20-%20All%20Branches/138/changes

**Comment by quartzjer** [comment]
this was fixed in node-clucene 0.2.2 :)
